app/controllers/shops_controller.rb

```diff
@@ -4,5 +4,11 @@ class ShopsController < BaseController
   before_filter :enable_embedded_shopfront
 
   def index
+    @enterprises = Enterprise
+      .activated
+      .includes(address: :state)
+      .includes(:properties)
+      .includes(supplied_products: :properties)
+      .all
```

Comment:
Nice @kristinalim ! Although on my list I never used `MemoryProfiler` before.

To be honest, I pulled my hair out when I see that `#all` call 😓 but I can't remove it yet until we add pagination. The current frontend implementation seems to depend on it to filter data on the client side.

Sadly, I think numbers are worse than what you tested... We're not loading all products but Enterprises with their addresses, states, properties and the properties of the products they supply 🙈 . We'll see if our app servers survive...

Reply:
> We're not loading all products but Enterprises with their addresses, states, properties and the properties of the products they supply see_no_evil .

But I had confirmed that it would keep all products in memory as well. :see_no_evil: Which is already in the thousands on some instances, with many attributes.

Reply:
@sauloperez It might be worth trying to replicate the enterprise (and address, state, average no. of properties) to product (and average no. of properties) ratio in a staging or even local server before putting this to production?

Reply:
Yes, we can. This will give us a good understanding of what to expect in production. @Matt-Yorkley would you mind setting up a production DB dump on UK's staging since it's already monitored in Datadog? I'd watch for mem usage besides the response time of those pages.